### PR TITLE
add material theme

### LINKED ISSUE #318: Add Material theme
https://github.com/Emiller88/emacs-doom-community-themes/pull/2/files

### PR DESCRIPTION
**Pull request by tam5**
Closes #318.

Adds initial material theme, which is mostly the same as palenight theme with default material background.

(Future plans - configurable accent color, other material varations)

### CODE REVIEW

**Comment by ema2159** [comment]
Could you upload a screenshot please?

**Comment by tam5** [comment]
Screen Shot 2019-12-11 at 7 41 40 PM

**Comment by ema2159** [comment]
Nice! I will test it as soon as possible.

**Comment by ema2159** [comment]
Are you using `(solaire-mode-swap-bg)` or taking it into account? It seems that you're assigning the `bg` color to the `bg-alt`'s and vice versa. The darkest background color is usually supposed to be `bg-alt`.

**Comment by tam5** [comment]
Actually @ema2159 I have two questions on that:

1. I did intentionally flip the colors, so that when solaire mode is turned off, the "correct" background is shown, which would be the same color as the editor BG. On the other hand, if solaire mode is on, the editor remains the same, but the extra buffers like treemacs get additional contrast.

2. Yes you are correct I should be using `solaire-mode-swap-bg` but I'm not sure where exactly that is supposed to go. I don't see in the palenight theme at all. It seems like the right place for it currently would actually be in here https://github.com/hlissner/doom-emacs/blob/develop/modules/ui/doom/config.el#L3-L23 somewhere. Anyway, please advise, thanks!

**Comment by ema2159** [comment]
1- I understand. The thing is that when I load the theme, the dark background is loaded in the main buffers and the light background is loaded in the secondary buffers (treemacs, which-key etc), on the contrary of the screenshot you posted (having `solaire-swap-bg` active).

2-`solaire-swap-bg` is external from the package, it goes into your config, and should always be active when using solaire-mode with doom-themes, as advised by hlissner.

File: themes/doom-material-theme.el [hunk 1]
```diff
@@ -0,0 +1,153 @@
...
+  :group 'doom-material-theme
+  :type '(choice integer boolean))
+
+;;
+(def-doom-theme doom-material
+  "A dark theme inspired by Material Theme by equinusocio"
+
+  ;; name        default   256       16
+  ((bg         '("#1C262B" nil       nil))
+   (bg-alt     '("#263238" nil       nil))
```
Comment (ema2159) [inline themes/doom-material-theme.el]: Switch these two (bg and bg-alt). That's would solve the problem.

**Comment by wpcarro** [comment]
Nice! Good work, everyone. Excited to use this theme.

**Comment by ema2159** [comment]
@tam5 awesome work! This implementation has become one of my favorite themes! I use it a lot for web dev because it allows me to focus incredibly well.

**Pull request merged**